integrations/tensorflow/e2e/keras/BUILD

```diff
@@ -0,0 +1,110 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load(
+    "//bindings/python:build_defs.oss.bzl",
+    "INTREE_TENSORFLOW_PY_DEPS",
+    "NUMPY_DEPS",
+)
+load(
+    "//build_tools/bazel:iree_py_test_suite.bzl",
+    "iree_py_test_suite",
+)
+load(
+    "//integrations/tensorflow/e2e/keras:iree_vision_test_suite.bzl",
+    "iree_vision_test_suite"
+)
+
+package(
+    default_visibility = ["//visibility:public"],
+    licenses = ["notice"],  # Apache 2.0
+)
+
+iree_py_test_suite(
+    name = "non_vision",
+    srcs = glob(
+        ["*_test.py"],
+        exclude = [
+            "keras_vision_model_test.py",
+            # TODO(GH-1620): Include after fixing the failure on GitHub Actions.
+            "keras_lstm_static_test.py"
+        ]
+    ),
+    deps = INTREE_TENSORFLOW_PY_DEPS + NUMPY_DEPS + [
+        "//integrations/tensorflow/bindings/python/pyiree/tf/support",
+    ],
+)
+
+iree_vision_test_suite(
+    name = "keras_vision_models",
+    configurations = [
+        # tuples of (dataset, include_top, model_name, backends)
+        # "cifar10" has toy models with input 32x32, is good for debugging
+        ("cifar10", 1, "ResNet50", "tf,iree_vmla"),
+        ("cifar10", 1, "ResNet50", "tf,iree_llvmjit"),
+        ("cifar10", 1, "ResNet50", "tf,iree_vulkan"),
+    ],
+    deps = INTREE_TENSORFLOW_PY_DEPS + NUMPY_DEPS + [
+        "//integrations/tensorflow/bindings/python/pyiree/tf/support",
+    ],
+    tags = ["manual"],
+)
+
+iree_vision_test_suite(
+    name = "keras_vision_models_external",
+    configurations = [
+        # tuples of (dataset, include_top, model_name, backends)
+        # "cifar10" has toy models with input 32x32, is good for debugging
+        # "imagenet" has real model weights for input 224x224
+        ("cifar10", 1, "MobileNet", "tf,iree_vmla"),
+        ("cifar10", 1, "MobileNet", "tf,iree_vulkan"),  # TODO(b/150244105)
+        ("cifar10", 1, "MobileNet", "tf,iree_llvmjit"),  # TODO(b/150244105)
+        ("cifar10", 1, "MobileNetV2", "tf,iree_vmla"),
+        ("cifar10", 1, "MobileNetV2", "tf,iree_vulkan"),  # TODO(b/150244105)
+        ("cifar10", 1, "MobileNetV2", "tf,iree_llvmjit"),  # TODO(b/150244105)
+        ("imagenet", 1, "ResNet50", "tf,iree_vmla"),
+        ("imagenet", 1, "ResNet50", "tf,iree_vulkan"),
+        ("imagenet", 1, "ResNet50", "tf,iree_llvmjit"),
+        ("imagenet", 1, "MobileNet", "tf,iree_vmla"),
+        ("imagenet", 1, "MobileNet", "tf,iree_vulkan"),  # TODO(b/150244105)
+        ("imagenet", 1, "MobileNet", "tf,iree_llvmjit"),  # TODO(b/150244105)
+        ("imagenet", 1, "MobileNetV2", "tf,iree_vmla"),
+        ("imagenet", 1, "MobileNetV2", "tf,iree_vulkan"),  # TODO(b/150244105)
+        ("imagenet", 1, "MobileNetV2", "tf,iree_llvmjit"),  # TODO(b/150244105)
+    ],
```

Comment:
So it looks like we're actually using the full combinatorial explosion here except for ResNet50 which is covered by the above (although not with external weights, so I think we could add it here). I think this could instead be

```suggestion
    # "cifar10" has toy models with 32x32 input which are good for debugging
    # "imagenet" has real model weights for 224x224 input
    datasets = ["cifar10", "imagenet"],
    reference_backend = "tf",
    test_backends = ["iree_vmla", "iree_vulkan", "iree_llvmjit"],
    models = ["ResNet50", "MobileNet", "MobileNetV2"],
```

Reply:
I think that this would make sense to consider as a separate change. There's also the question of how we would map those bugs do the correct configurations. At the very least I think that we should ask Oleg.

Reply:
Ok I'm fine with it as a separate change if you want. The advantage of a macro is it allows you to do things like this that you can't with a list comp. But the macro breaks things out into a separate function and file, which puts a higher burden on having good abstractions so that the usage is clear at the call site.